File: src/leetCodeProblems/TwoPointers/LengthOfLastWord58.go
package main

/*
- LeetCode - https://leetcode.com/problems/length-of-last-word/
- Time - O(n)
- Space - O(1)
*/
func lengthOfLastWord(s string) int {
	start, end, endFound := -1, -1, false

	for i := len(s) - 1; i >= 0; i-- {

		// endFound is not found & non-empty character is found
		if !endFound && string(s[i]) != " " {
			end = i
			endFound = true
		}

		// endFound is found & empty character is found
		if endFound && string(s[i]) == " " {
			start = i
			break
		}
	}
	return end - start
}

File: src/leetCodeProblems/TwoPointers/LengthOfLastWord58.java
package leetCodeProblems.BruteForce;

/**
 * LeetCode - https://leetcode.com/problems/length-of-last-word/
 */

public class 58 {

    public int (String s) {

        String[] input = s.split(" ");
        return input[input.length-1].length();
    }

    public static void main(String[] args) {
        58 obj = new 58();
        System.out.println(obj.("   fly me   to   the moon  "));
    }
}
